### ISSUE TITLE
TAB order of app

### ISSUE THREAD

**Issue by CKFinderBOT** [ckfinder/ckfinder#288]
taking starting point from: #936:

On current master it work probably the same, but one minor thing get fixed (focus on whole page) which was not visible. Now the tab order of the important things in app is done based on `tabindex` value, which apparently browsers treats a differently.

On IE10: when focused element has `tabindex="-1"` (it does not take part in tab order but is focusable, ie toolbar buttons, files, folders) and user hit tab the focus goes like this:
TAB: address bar > compatibility view in address bar > current tab > home button right to address bar > then first item with positive `tabindex` on page (toolbar) > filter > folders > files pane > address bar and so on
SHIFT-TAB in reverse order

On Chrome: toolbar > filter > folders > files pane > address bar > toolbar and so on
SHIFT-TAB in reverse order
When element has `tabindex=-1` the ordering depends on element placement inside DOM, so one would have to check if order of toolbar, folders panel and files pane is as expected (I think that you can get unexpected results when going TAB or SHIFT-TAB when focused on file / folder.

On Firefox: 
start is the same order as in Chrome, but goes through search bar and buttons groups in toolbar,

**Important** Tab order start point is relative to your cursor when entering page, ie:
- doing refresh when cursor is inside will start tabbing from address bar
- doing refresh when focus is inside CKFinder then after refresh you start as described above.

This bring some incosistancy to TAB order, so the question is:
- do we care on those?

If not - problem solved (check the DOM order of elements).
If yes - we should introduce some kind of API for registering TAB order priorities of some DOM elements and defining, ie:

```js
request( 'ui:tab:register', { name: 'toolbar', priority: 10 } );
request( 'ui:tab:register', { name: 'folders', priority: 20 } );
request( 'ui:tab:register', { name: 'files', priority: 30 } );

/// now imagin user press tab on some ui element:
// this will be done in view
this.finder.request( 'ui:tab:next', { current: 'files' } );
// or prev:
this.finder.request( 'ui:tab:prev', { current: 'files' } );

// finder will check wich tab element to focus and fire event:
this.fire( 'ui:tab:focus:toolbar' );

// so any view can listen to it and focus itself:
finder.on( 'ui:tab:focus:toolbar', function( 
    this.$el.focus();
} );
```

This will add extra work for modules then defining `tabindex` inside DOM. Also is problematic on dialogs, other views.

As a variant we might only register DOM nodes to this module, but this will require to re-define a node after rendering a view.

```js
request( 'ui:tab:register', { name: 'toolbar', priority: 10, el: this.$el } );
// now finder will listen to keydown and move focus by itself (no need to do anything from module)

el.on( 'keydown', function( evt ) {
    // check current item name, get next if tab or pref if shift tab and then
   getNextItem( currentName ).focus();
} );

// now any view would call `'ui:tab:register'` on each `render`.
```
Above slightly less work ,I think that almost the same problematic , and also require some work for dialogs & stuff

**Comment by CKFinderBOT** [comment]
Fixed. The updated code will be publicly available in the upcoming release.
